matminer/featurizers/composition/roost.py

```diff
@@ -0,0 +1,569 @@
+            )
+
+        if n_targets is None:
+            if task == "regression":
+                n_targets = 1  # hard coded for single target regression
+            elif task == "classification":
+                n_targets = 2  # hard coded for binary clf
+
+        if elem_emb == "matscholar":
+            elem_emb_len = 200  # hard coded for matscholar
+        else:
+            raise NotImplementedError(
+                "Currently only the matscholar embedding is implemented"
+            )
+
+        self.data_params = {
+            "batch_size": batch_size,
+            "num_workers": workers,
+            "pin_memory": False,
+            "shuffle": True,
+            "collate_fn": collate_batch,
+        }
+
+        self.setup_params = {
+            "loss": loss,
+            "optim": optim,
+            "learning_rate": learning_rate,
+            "weight_decay": weight_decay,
+            "momentum": momentum,
+            "device": device,
+        }
+
+        self.model_params = {
+            "task": task,
+            "robust": False,
+            "n_targets": n_targets,
+            "elem_emb_len": elem_emb_len,
+            "elem_fea_len": elem_fea_len,
+            "n_graph": n_graph,
+            "elem_heads": elem_heads,
+            "elem_gate": elem_gate,
+            "elem_msg": elem_msg,
+            "cry_heads": cry_heads,
+            "cry_gate": cry_gate,
+            "cry_msg": cry_msg,
+            "out_hidden": out_hidden,
+        }
+
+        model, criterion, optimizer, scheduler, normalizer = init_roost(
+            **self.setup_params,
+            **self.model_params,
+        )
+
+        self.model_name = model_name
+        self.run_id = run_id
+
+        self.model = model
+        self.criterion = criterion
+        self.optimizer = optimizer
+        self.scheduler = scheduler
+        self.normalizer = normalizer
+
+        self.model_name = model_name
+        self.run_id = run_id
+
+        self.elem_fea_len = elem_fea_len
+        self.elem_emb = elem_emb
+        self.task = task
+        self.epochs = epochs
+
+        self.fitted = False
+
+        # NOTE multiprocessing doesn't play nicely with pytorch, apparently you
+        # can set the multiprocessing start method to spawn but it's safer just
+        # to only use one job?
+        # set_start_method('spawn', force=True)
+        self.set_n_jobs(1)
+
+    def fit(self, X, y, checkpoint=False, log=False, verbose=False):
+
+        train_set = CompositionData(X, targets=y, task=self.task, emb=self.elem_emb)
+
+        # NOTE this is needed as we declare the model upfront without knowledge of the
+        # dataset.
+        if train_set.n_targets != self.model_params["n_targets"]:
+            raise ValueError("Number of targets for dataset does not match model")
+
+        train_generator = DataLoader(train_set, **self.data_params)
+
+        # if val_set is not None:
```

Comment:
We don't have any guidelines for validation sets. Do as you wish!

Reply:
I will remove this then as partitioning training data would hurt the model performance and in my testing I haven't yet encountered examples where early stopping would be significant. As above I think if someone wants to do an extensive search over hyperparameters I think it would be easier outside of `matminer`.